### ISSUE TITLE
Remove deprecated code in twisted.python.log

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5765]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#5765|
|Type|task|
|Created|2012-07-11 10:21:38Z|
|Branch|https://github.com/twisted/twisted/tree/remove-deprecated-log-functions-5765|

The following methods in twisted.python.log are deprecated since Twisted 2.5 and can be removed:

 * `startKeepingErrors`
 * `flushErrors` and `_flushErrors`
 * `ignoreErrors` and `_ignore`
 * `clearIgnores` and `_clearIgnores`

Attachments:

* [log-5765.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/2c8/2c8f75e6a1d41dba3e032adab0c39701d3d610e6/b15ba3f904a9addb799669c787ce58beb58e9c1c.patch) (4303 bytes) - added by thijs on 2012-07-12 12:09:50Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5765 5765
type__task task
reporter__thijs thijs
priority__low low
milestone__ 
branch__branches_remove_deprecated_log_functions_5765 branches/remove-deprecated-log-functions-5765
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1342002098000000 1342002098000000
changetime__1342334893000000 1342334893000000
version__None None
owner__spiv spiv
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv set owner to spiv<br>spiv set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|spiv set owner to @thijstriemstra<br>spiv set status to `new`|
|-|-|

I don't understand why `twisted.trial.unittest._LogObserver._add` still assigns to `log._flushErrors` etc.  Isn't the point of this patch that `log._flushErrors` and friends don't exist anymore?

If those lines were deleted, this patch looks fine to me.

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|spiv set owner to spiv<br>spiv set status to `assigned`|
|-|-|

As I expected, tests still pass without those lines.  I'll make that change and put it up for review.

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv removed owner<br>spiv set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_8"></a>|@cyli set owner to spiv|
|-|-|

Thanks for reviewing and fixing this spiv!  I was wondering about those too - this looks good to me.  Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_9"></a>|@cyli commented|
|-|-|

(also, [buildbot passes](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/remove-deprecated-log-functions-5765))

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_10"></a>|spiv set status to `closed`|
|-|-|

(In [34813]) Merge remove-deprecated-log-functions-5765: Removes deprecated functions from twisted.python.log.

Authors: thijs, spiv
Reviewers: spiv, cyli
Fixes: [#5765](https://github.com/twisted/twisted/issues/5765)

Removes startKeepingErrors, flushErrors, _flushErrors, ignoreErrors, _ignore,
clearIgnores and _clearIgnores from twisted.python.log.  They've been
deprecated since Twisted 2.5.
